docs/topics/wasm/wasm-overview.md

```diff
@@ -1,61 +1,75 @@
+Kotlin/Wasm as its compilation target.
 
-> Learn more about Kotlin/Wasm with this [YouTube playlist](https://kotl.in/wasm-pl).
+Explore our online demo of an application built with Compose Multiplatform for Web and Kotlin/Wasm:
+
+<a href="https://zal.im/wasm/jetsnack/"><img src="wasm-demo.png" width="700" alt="Kotlin/Wasm demo"/></a>
+
+>To run applications built with Kotlin/Wasm in a browser, you need a browser version that supports the new garbage collection
+> and exception handling proposals. To check the support status, see the [WebAssembly
+> roadmap](https://webassembly.org/roadmap/).
 >
-{type="note"}
+{type="tip"}
 
-## Browser support
+[WebAssembly (Wasm)](https://webassembly.org/) is a binary instruction format for a stack-based virtual machine. This
+format is platform-independent because it runs on its own virtual machine. Wasm provides languages like Kotlin, and many 
+more, with a compilation target so that they can run on the web.
 
-To run applications built with Kotlin/Wasm in a browser, you need a version supporting the new [garbage collection feature](https://github.com/WebAssembly/gc).
+Kotlin/Wasm compiles your Kotlin code into Wasm format. Using Kotlin/Wasm, you can create applications that run on 
+different environments and devices, which support Wasm and meet Kotlin's requirements.
 
-[Learn more in Get started with Kotlin/Wasm](wasm-get-started.md#troubleshooting).
+Would you like to try it yourself?
 
-## Interoperability
+<a href="wasm-get-started.md"><img src="wasm-get-started-button.png" width="700" alt="Get started with Kotlin/Wasm"/></a>
 
-Kotlin/Wasm allows you to both use JavaScript code and Browser API from Kotlin, and Kotlin code from JavaScript.
+## Performance
 
-[Learn more about Kotlin Wasm interoperability with JavaScript](wasm-js-interop.md).
+Although Kotlin/Wasm is still in Alpha, Compose Multiplatform running on Kotlin/Wasm already shows encouraging performance 
+traits. You can see that the execution speed outperforms JS and is getting close to JVM:
 
-## Compose Multiplatform for Web
+![Kotlin/Wasm performance](wasm-performance-compose.png){width=700}
 
-> Web support is in [Alpha](components-stability.md) and may be changed at any time. Use it only for evaluation purposes.
-> We would appreciate your feedback on it in the public Slack channel [#compose-web](https://slack-chats.kotlinlang.org/c/compose-web).
-> If you face any issues, please report them on [GitHub](https://github.com/JetBrains/compose-multiplatform/issues).
->
-{type="note"}
+We regularly run benchmarks on Kotlin/Wasm, and these results come from our testing in a recent version of Google Chrome.
+
+## Browser API support
+
+The Kotlin/Wasm standard library provides declarations for browser APIs (including the DOM API).
+With these declarations, you can directly use the Kotlin API to access and utilize various browser functionalities, like 
```

Comment:
Well, I'd like to see this one a bit simplified. Maybe we should divide it into two parts?